bindingcollectionadapter/src/main/java/me/tatarka/bindingcollectionadapter2/itembindings/OnItemBindClass.java

```diff
@@ -44,6 +74,26 @@ public int itemTypeCount() {
 
     @Override
     public void onItemBind(ItemBinding itemBinding, int position, T item) {
+        itemBinding.clearExtra();
+        if (itemExtraBindingMap != null && !itemExtraBindingMap.isEmpty()) {
+            for (int i = 0; i < itemExtraBindingMap.size(); i++) {
+                Class<? extends T> key = itemExtraBindingMap.keyAt(i);
+                if (key.isInstance(item)) {
```

Comment:
Sorry about that! I think I forgot to update the javadoc after changing it's behavior. I found an `isInstance()` check makes it easier to work with autovalue since you dont' have to supply the generated class.